src/layout/IHyprLayout.cpp

```diff
@@ -178,6 +182,54 @@ void IHyprLayout::onWindowCreatedFloating(PHLWINDOW pWindow) {
     }
 }
 
+bool IHyprLayout::onWindowCreatedAutoGroup(PHLWINDOW pWindow) {
+    static auto AUTOGROUP = CConfigValue<Hyprlang::INT>("group:auto_group");
+    if ((*AUTOGROUP || g_pInputManager->m_bWasDraggingWindow) && g_pCompositor->m_pLastWindow.lock() && g_pCompositor->m_pLastWindow != pWindow &&
```

Comment:
some comments here would help clear up the hellish if

Reply:
is this ok?
```
if ((*AUTOGROUP || g_pInputManager->m_bWasDraggingWindow) // check if auto_group is enabled or instead if the user is manually dragging the window into the group
    && g_pCompositor->m_pLastWindow.lock() // check if a focused window exists
    && g_pCompositor->m_pLastWindow != pWindow // fixes toggling a floating group into a tiled group
    && g_pCompositor->m_pLastWindow->m_pWorkspace == pWindow->m_pWorkspace // fix for multimonitor: when there is a focused group in monitor 1 and monitor 2 is empty, this enables adding the first window into monitor 2 when using the mouse to focus it
    && g_pCompositor->m_pLastWindow->m_sGroupData.pNextWindow.lock() // check if the focused window is a group
    && pWindow->canBeGroupedInto(g_pCompositor->m_pLastWindow.lock()) // check if the new window can be grouped into the focused window
    && !g_pXWaylandManager->shouldBeFloated(pWindow)) { // don't group XWayland windows that should be floated.
```